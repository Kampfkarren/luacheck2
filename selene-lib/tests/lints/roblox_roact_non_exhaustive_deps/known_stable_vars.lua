local React

local function Component()
    local shouldNotIgnoreState, shouldIgnoreSetState = React.useState()
    local shouldIgnoreBinding, shouldIgnoreSetBinding = React.useBinding()
    local shouldIgnoreRef = React.useRef()

    React.useEffect(function()
        print(shouldNotIgnoreState, shouldIgnoreSetState)
        print(shouldIgnoreBinding, shouldIgnoreSetBinding)
        print(shouldIgnoreRef)
    end, {})

    React.useEffect(function()
        print(shouldIgnoreSetState)
        -- Even though passing known stable variables isn't needed, it shouldn't warn about it
    end, { shouldIgnoreSetState })
end
