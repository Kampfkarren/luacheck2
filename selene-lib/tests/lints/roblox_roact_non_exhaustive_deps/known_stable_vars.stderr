error[roblox_roact_non_exhaustive_deps]: react hook useEffect has missing dependency: 'shouldNotIgnoreState'
   ┌─ known_stable_vars.lua:12:10
   │
12 │     end, {})
   │          ^^
   │
   = help: either include it or remove the dependency array

